src/pages/admin/UserManager.js

```diff
@@ -1,275 +1,242 @@
-  }
+
+    setUsers(newusers);
+    setFilteredUsers(newfiltered_users);
+  };
 
   /**
    * Function to apply filter to user data
    * @param {Value of Filter} filterValue
    * @param {Search Filter to apply search on} filterType
    */
-  updateSearchFilter(filterValue, filterType) {
-    // Update the state with the filter value before filtering the data
-    this.setState(
-      {
-        [filterType]: filterValue,
-      },
-      () => {
-        const stateObj = { ...this.state };
-        // Apply the search filters to original users data
-        // TODO: room for optimize by filtering on existing filtered data
-        const filteredUsers = this.state.users.filter((user) => {
-          const searchTextFilter = stateObj.searchText
-            ? user.name
-              .toLowerCase()
-              .includes(stateObj.searchText.toLowerCase())
-              || user.email
-                .toLowerCase()
-                .includes(stateObj.searchText.toLowerCase())
-              || user.number
-                .toLowerCase()
-                .includes(stateObj.searchText.toLowerCase())
-            : true;
-          const userStatusFilter = stateObj.userStatusFilter
-            ? user.status === stateObj.userStatusFilter
-            : true;
-          const locationFilter = stateObj.locationFilter
-            ? user.state === stateObj.locationFilter
-              || user.city === stateObj.locationFilter
-            : true;
-          const userTypeFilter = stateObj.userTypeFilter
-            ? user.type === stateObj.userTypeFilter
-            : true;
-          return (
-            searchTextFilter
-            && userStatusFilter
-            && locationFilter
-            && userTypeFilter
-          );
-        });
-
-        // Update the filtered data in the state
-        this.setState({
-          filteredUsers,
-        });
-      },
-    );
-  }
-
+  const updateSearchFilter = (filterValue, filterType) => {
+    setFilters({...filters,[filterType]: filterValue})
+  };
+useEffect(()=>{
+  const filteredList = users.filter((user) => {
+    const searchTextFilter  = filters.searchText
+      ? user.name
+        .toLowerCase()
+        .includes(filters.searchText?.toLowerCase())
+        || user.email
+          .toLowerCase()
+          .includes(filters.searchText?.toLowerCase())
+        || user.mobileNumber
+          .toLowerCase()
+          .includes(filters.searchText?.toLowerCase())
+      : true;
+    const userStatus = filters.userStatusFilter
+      ? user.userStatus === filters.userStatusFilter
+      : true;
+    const location = filters.locationFilter
+      ? user.state === filters.locationFilter
+        || user.city === filters.locationFilter
+      : true;
+    const userType = filters.userTypeFilter
+      ? user.userRole === filters.userTypeFilter
+      : true;
+      return (searchTextFilter && userStatus && location && userType)
+  });
+  setFilteredUsers(filteredList);
+},[filters])
```

Comment:
Instead of using use effect to check if a filter is updated we should have event listener (eg onChange listener for a change event) on which we can execute this code. useEffect is generally used when the state is a prop or outside the control of a component. Also, we should avoid if possible having multiple useEffect in a single component.